src/lib/Zikula/Core/Theme/AssetBag.php

```diff
@@ -19,18 +39,22 @@ public function __construct(array $assets = array())
      */
     public function add($asset)
     {
+        // ensure value is an array
         if (!is_array($asset)) {
-            $asset = array($asset);
+            $asset = [$asset => self::DEFAULT_WEIGHT];
         }
-        $this->assets = array_merge($this->assets, $asset);
-        $this->assets = array_unique($this->assets);
+        foreach ($asset as $source => $weight) {
+            if ((isset($this->assets[$source]) && $this->assets[$source] > $weight) || !isset($this->assets[$source])) {
+                // keep original weight if lighter. set if not set already.
+                $this->assets[$source] = $weight;
+            }
+        }
+        asort($this->assets); // put array in order by weight
```

Comment:
I disagree that this is a problem. if order matters to you, set the weight to something else. we don't need a complicated solution to it. also see the [class docblock](https://github.com/zikula/core/pull/2606#discussion-diff-40910160L12)